src/data/nat/nth.lean

```diff
@@ -0,0 +1,406 @@
+/-
+Copyright (c) 2021 Vladimir Goryachev. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Vladimir Goryachev, Kyle Miller, Scott Morrison, Eric Rodriguez
+-/
+import data.list.basic
+import data.nat.prime
+import data.nat.count
+import order.order_iso_nat
+import set_theory.fincard
+
+/-!
+# The `n`th Number Satisfying a Predicate
+
+This file defines a function for "what is the `n`th number that satisifies a given predicate `p`",
+and provides lemmas that deal with this function and its connection to `nat.count`.
+
+## Main definitions
+
+* `nth p n`: The `n`-th natural `k` (zero-indexed) such that `p k`. If there is no
+  such natural (that is, `p` is true for at most `n` naturals), then `nth p n = 0`.
+
+## Main results
+
+* `nat.nth_eq_order_iso_of_nat`: An infinite set of natural numbers is order-isomorphic to the
+  natural numbers.
```

Comment:
This docstring is out of date.

Reply:
How so?

Reply:
I guess you want me to mention some other main results, so I'll list a few that seem important to me.

Reply:
Sorry, I misread the docstring. I should've said that the lemma claims that two maps are equal, not that there exists an order isomorphism. Otherwise LGTM.
bors d+

Reply:
Ah yes, I see my mistake.